gslib/utils/system_util.py

```diff
@@ -250,11 +249,19 @@ def IsRunningInteractively():
 
 
 def MonkeyPatchHttp():
-  if platform.python_version().startswith('3.6.5'):
-    _MonkeyPatchHttpForPython_3_6_5()
-
-
-def _MonkeyPatchHttpForPython_3_6_5():
+  # Getting python version array in the following format: [major, minor, micro]
+  py_ver = [
+      sys.version_info.major,
+      sys.version_info.minor,
+      sys.version_info.micro
+  ]
+  # Checking for and applying monkeypatch to python versions 3.4 - 3.6.7
+  if (py_ver[0] == 3 and
+      4 <= py_ver[1] < 6 or (py_ver[1] == 6 and py_ver[2] < 8)):
```

Comment:
Sorry if I was unclear! My nit before mainly applied here, where it would be easier to read by not using list indexes for comparison :)

```
import sys.version_info as ver
if ver.major == 3:
    if ver.minor < 6 or (ver.minor == 6 and ver.micro < 8):
        _MonkeyPatchHttpForPython_3x()
```

Reply:
Any PY3 version below v3.6.7 should have our monkeypatch

Reply:
changed